src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -106,6 +106,8 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool isBackgroundSync
             {
                 await _syncSemaphore.WaitAsync();
 
+                PrepareSyncTriggers();
```

Comment:
I'm basically reverting the revert done in https://github.com/Azure/azure-functions-host/pull/7915/files here.